src/lib/asn1/oid_map.cpp

```diff
@@ -51,6 +51,10 @@ void OID_Map::add_oid2str(const OID& oid, std::string_view str) {
 }
 
 std::string OID_Map::oid2str(const OID& oid) {
+   if(auto name = lookup_static_oid(oid)) {
+      return *name;
+   }
+
```

Comment:
As mentioned in the comment on `lookup_static_oid`, I think we should postpone the allocation of `std::string` until here. 

Or, if safe, let this function return a string view as well. Consequently, that would also return string views onto user-defined OID registrations below. Though, that should be safe as long as we don't allow the user to deregister anything.

Reply:
Mhh... strings should be moved. So even if the `unordered_map` re-allocs, this _should_ be fine. I'm not sure how short-string-optimization plays into this. Better safe than sorry, I guess.

Reply:
> that should be safe as long as we don't allow the user to deregister anything.

We do not (any more) allow users to remove/delete OIDs (which is what makes this static lookup viable in the first place). So if we don't end up with dangling references even after an unsorted_map resize, we can return string_views. Which would be nice for sure, though TBH oid->string is *typically* just for debugging/error reporting, whereas string->oid happens more often in the course of normal applications.

[We could consider interning the common OIDs such that instead of a dynamically allocated vector they can just point directly at the static data. I'm not sure if this is worthwhile though...]